src/components/Form/FormItem.tsx

```diff
@@ -0,0 +1,115 @@
+import { Form as AntForm, FormItemProps as AntFormItemProps, FormInstance } from 'antd'
+import { CSSProperties, ReactElement, ReactNode, isValidElement, useMemo } from 'react'
+
+import { Checkbox } from '../Checkbox'
+import { Input } from '../Input'
+import { RadioGroup } from '../RadioGroup'
+import { Switch } from '../Switch'
+
+type RenderProps = {
+  form: FormInstance,
+  value: unknown;
+  onChange: (value: unknown) => void
+}
+export type FormItemProps = {
+  name?: string
+  label?: string
+  style?: CSSProperties
+  span?: number
+  isFullWidth?: boolean,
+  rules?: AntFormItemProps['rules']
+  valuePropName?: AntFormItemProps['valuePropName']
+  getValueFromEvent?: AntFormItemProps['getValueFromEvent']
+  children: ReactElement | ((props: RenderProps) => ReactNode)
+}
+
+const defaults = {
+  span: 1,
+  isFullWidth: false,
+}
+
+const getDefaultFormItemProps
+  = ({ type }: ReactElement): Partial<FormItemProps> => {
+    switch (type) {
+    case Input:
+      return {
+        getValueFromEvent: (_, value) => value,
+      }
+    case Checkbox:
+      return {
+        valuePropName: 'isChecked',
+        getValueFromEvent: ({ target }) => target.checked,
+      }
+    case Switch:
+      return {
+        valuePropName: 'isChecked',
+      }
+    case RadioGroup:
+      return {
+        getValueFromEvent: ({ value }) => value,
+      }
+    default:
```

Comment:
is this reachable? If we know all the possible types, I would error in case of unknown ones

Reply:
Yes, this is actually reachable: the element can have an unknown type if the input is implemented outside the library, so the __getDefaultFormItemProps__ should return an empty object (there are no known defaults to apply)